backends/dpdk/dpdkAsmOpt.h

```diff
@@ -159,6 +159,100 @@ class RemoveUnusedMetadataFields : public Transform {
     const IR::Node* preorder(IR::DpdkAsmProgram *p) override;
 };
 
+// This pass shorten the Identifier length
+class ShortenTokenLength : public Transform {
+    ordered_map<cstring, cstring> newNameMap;
+    ordered_map<cstring, cstring> &origNameMap;
+    static size_t count;
+    // Currently Dpdk allows Identifier of 63 char long or less
+    // including dots(.) for member exp.
+    // worst case member expression will look like
+    // 1.30.30 => 63(including dot(.))
+    cstring shortenString(cstring str, size_t allowedLength = 30) {
+        // if id name less than allowedLength keep it same
+        if (str.size() <= allowedLength)
+            return str;
+        auto itr = newNameMap.find(str);
+        if (itr != newNameMap.end())
+            return itr->second;
+	// make sure new string length less or equal allowedLength
+        cstring newStr = str.substr(0, allowedLength - std::to_string(count).size());
+        newStr += std::to_string(count);
+        count++;
+        newNameMap.insert(std::pair<cstring, cstring>(str, newStr));
+        origNameMap.insert(std::pair<cstring, cstring>(newStr, str));
+        return newStr;
+    }
+
+ public:
+    ShortenTokenLength(ordered_map<cstring, cstring> &origNameMap)
+        : origNameMap{origNameMap} {}
+    const IR::Node* preorder(IR::Member *m) override {
+        m->member = shortenString(m->member);
```

Comment:
We should consider the total length including m->expr.

Reply:
suppose total length is more than 63 , and it;s in form h.member1.member2, then which member will be changed member1 or member2?

Reply:
member2 only should change but it should change only when the total length exceeds the length

Reply:
what if member1 itself is longer than allowed(60), changing member2 won't help.